Clarify expectations of `flint_set_throw`'s argument (#2387)
flintlib/flint: doc/source/flint.rst

.. function:: void flint_set_throw(void (* func)(flint_err_t, const char *, va_list))

    Sets the :func:`flint_throw` function use ``func`` instead of a private

>     throw function. ``func`` is expected to ``va_end`` the ``va_list``,
>     but it does not take ownership of the ``char *``.

Reviewer: ```suggestion
    throw function. ``func`` is expected to call ``va_end``, and to not alter
    `const char *` in any way.
```

Author: Thanks, sounds good!
